src/convolution.jl

```diff
@@ -47,6 +50,20 @@ end
 convolve(d1::Poisson, d2::Poisson) =  Poisson(d1.λ + d2.λ)
 
 
+
+function convolve(d1::DiscreteNonParametric, d2::DiscreteNonParametric)
+    support_conv = vcat([d1.support .+ v for v in d2.support]...) |> unique 
+    sort!(support_conv) #for fast index finding below
+    p_conv = zeros(length(support_conv)) 
+    for (s1,p1) in zip(d1.support, d1.p)
+        for (s2,p2) in zip(d2.support, d2.p)
+            idx = searchsortedfirst(support_conv, s1+s2)
```

Comment:
I wonder if this (and the `unique` above) is problematic due to floating point inaccuracies.

Reply:
It shouldn't not matter as long as we do the same operations s1+s2 exactly both in creation os support_conv and in finding idx. 
Not sure how the compiler optimization might matter in this. 

Added a test with eps(Float64) to see any trivial inconsistencies.